Sources/App/Models/Meetup.swift

```diff
@@ -8,6 +8,13 @@
 import FluentPostgreSQL
 import Vapor
 
+enum Status: Int {
+    case scheduled = 1
+    case done
+    case cancelled
+    case inProgress
+}
+
 struct Meetup: PostgreSQLModel {
     
     static let IdKey: IDKey = \.id
```

Comment:
how an Id key is optional? also, instead of using Int shouldn't you use UUID as a better ID type? this applies for every model btw

Reply:
IdKey its just a typealias of a keypath (<Self,ID>),  where ID it's a associated type that needs an type that conforms to ID protocol.  yeah I agree I'll change to UUID.

Reply:
nvm, for some reason PostgreSQLModel protocol wants the id int and optional, idk why.

Reply:
according to the docs, it is possible to use Int, UUID, or String

I added the comments in this line but it applies to everything after this line

I think this may be because you are using structs and you can't really instantiate them with let properties without values, and this may be set after saved, not before

Reply:
yeah it's possible use UUID but needs to be optional. This is the answer that Tanner Nelson give to me when I asked about the optional ID

`t's optional because you should be able to let the database engine generate an ID upon saving
If it were non-optional, then you would always be required to set a value to initialize a model (even a newly created one that hasn't been saved yet)
There is a method requireID() to make it easier to access the value as a non-optional`